packages/rest/src/http/metadata.ts

```diff
@@ -245,6 +246,10 @@ function isApplicableMetadataCore(
   treatBodyAsMetadata: boolean,
   isMetadataCallback: (program: Program, property: ModelProperty) => boolean
 ) {
+  if (isKey(program, property)) {
+    return true;
+  }
+
```

Comment:
This will make the @key property disappear from responses where it can't become a path param. Is that what we want?

https://cadlplayground.z22.web.core.windows.net/prs/1304/?c=aW1wb3J0ICJAY2FkbC1sYW5nL3Jlc3QiOwoKQHNlcnZpY2UoewogIHRpdGxlOiAiV2lkZ2V0IFPGGyIsCn0pCm5hbWVzcGFjZSBEZW1vxxs7CnVzaW5nIENhZGwuSHR0cDsKCm1vZGVsIMdDxFZAa2V5IGlkOiBzdHJpbmc7CiAgd2VpZ2h0OiBpbnQzMjsKfQoKQHJvdXRlKCIvIikKb3AgdGVzdCguLi7GRyk6x1A7Cg%3D%3D

I think it would be better if isMetadata returned true for `@key`.

Reply:
The canonical case for HTTP/azure services is that the `@key` (and parent `@key`s) is/are a path parameter(s) for requests (and not in the response-body) and (read-only property/ies) in response bodies. There are some services (e.g. ARM) where the `@key` property is not in the response, and sometimes (if there is not composition), only item keys and not parent keys are in the response.

Reply:
@nguerrera Not intended, thanks for the catch!  I'll try your suggestion.